ensime_shared/ensime.py

```diff
@@ -771,32 +33,36 @@ class Ensime(object):
     There is normally one instance of ``Ensime`` per Vim session. It manages
     potentially multiple ``EnsimeClient`` instances if the user edits more than
     one ENSIME project.
+
+    Args:
+        vim: The ``vim`` module/singleton from the Vim Python API.
+
+    Attributes:
+        clients (Mapping[str, EnsimeClient]):
+            Active client instances, keyed by the filesystem path to the
+            ``.ensime`` configuration for their respective projects.
     """
 
     def __init__(self, vim):
-        self.vim = vim
-        # Map ensime configs to a ensime clients
+        # NOTE: The vim object cannot be used within the constructor due to
+        # race condition of autocommand handlers being invoked as they're being
+        # defined.
+        self._vim = vim
         self.clients = {}
 
-    def init_settings(self):
-        """Loads all the settings from the ``g:ensime_*`` namespace.
-
-        Invoked on client creation to avoid ``autocmd`` deadlocks.
-        """
-        self.server_v2 = bool(self.get_setting('server_v2', 0))
+    def using_server_v2(self):
+        """Whether user has configured the plugin to use ENSIME v2 protocol."""
+        return bool(self.get_setting('server_v2', 0))
```

Comment:
After giving it thought (prompted by pylint which raises an eyebrow at instance attributes defined outside of `__init__`), I thought an accessor-type function here was nicer at least while there's only one setting. It avoids the question of where/when `init_settings` gets called due to the matter of `vim` use in the constructor, and it could also allow for switching dynamically between server versions without restarting Vim, which could be useful for development especially after we add some server lifecycle management.